pkl-commons-test/pkl-commons-test.gradle.kts

```
@@ -16,116 +16,113 @@ dependencies {
     isReproducibleFileOrder = true
   }
 
-  val copyTask = tasks.create("copy-${packageDir.name}", Copy::class) {
-    dependsOn("zip-${packageDir.name}")
+  val copyTask = tasks.register("copy-${packageDir.name}", Copy::class) {
+    dependsOn(zipTask)
     from(metadataJson)
     into(destinationDir)
     val shasumFile = file("$destinationDir/${packageDir.name}.json.sha256")
     outputs.file(shasumFile)
-    doFirst {
-      expand(mapOf("computedChecksum" to archiveFile.computeChecksum()))
+    filter { line ->
+      line.replaceFirst("\$computedChecksum", archiveFile.computeChecksum())
     }
     doLast {
-      val outputFile = file("$destinationDir").resolve("${packageDir.name}.json")
+      val outputFile = destinationDir.resolve("${packageDir.name}.json")
       shasumFile.writeText(outputFile.computeChecksum())
     }
-    createTestPackages.dependsOn(this)
   }
 
-  sourcesJar.dependsOn.add(copyTask)
+  createTestPackages.configure { 
+    dependsOn(copyTask) 
+  }
 }
 
+val keystoreDir = file("build/keystore")
+val keystoreName = "localhost.p12"
+val certsFileName = "localhost.pem"
+
 val generateKeys by tasks.registering(JavaExec::class) {
-  val outputFile = file("$buildDir/keystore/localhost.p12")
+  val outputFile = file("$keystoreDir/$keystoreName")
   outputs.file(outputFile)
   mainClass.set("sun.security.tools.keytool.Main")
   args = listOf(
     "-genkeypair",
     "-keyalg", "RSA",
     "-alias", "integ_tests",
-    "-keystore", "localhost.p12",
+    "-keystore", keystoreName,
     "-storepass", "password",
     "-dname", "CN=localhost"
   )
-  workingDir = file("$buildDir/keystore/")
-  onlyIf { !outputFile.exists() }
+  workingDir = keystoreDir
   doFirst {
     workingDir.mkdirs()
+    outputFile.delete()
   }
 }
 
-val generateCerts by tasks.registering(JavaExec::class) {
-  dependsOn("generateKeys")
-  val outputFile = file("$buildDir/keystore/localhost.pem")
+val exportCerts by tasks.registering(JavaExec::class) {
+  val outputFile = file("$keystoreDir/$certsFileName")
+  dependsOn(generateKeys)
+  inputs.file("$keystoreDir/$keystoreName")
   outputs.file(outputFile)
   mainClass.set("sun.security.tools.keytool.Main")
   args = listOf(
     "-exportcert",
     "-alias", "integ_tests",
     "-storepass", "password",
-    "-keystore", "localhost.p12",
+    "-keystore", keystoreName,
     "-rfc",
-    "-file", "localhost.pem"
+    "-file", certsFileName
   )
-  workingDir = file("$buildDir/keystore/")
-  onlyIf { !outputFile.exists() }
+  workingDir = keystoreDir
   doFirst {
     workingDir.mkdirs()
+    outputFile.delete()
   }
 }
+
+fun toHex(hash: ByteArray): String {
+  val hexDigitTable = charArrayOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f')
+  val builder = StringBuilder(hash.size * 2)
+  for (b in hash) {
+    builder.append(hexDigitTable[b.toInt() shr 4 and 0xF])
+    builder.append(hexDigitTable[b.toInt() and 0xF])
+  }
+  return builder.toString()
+}
```

While we are here changing things:

```suggestion
fun toHex(hash: ByteArray): String {
    val hexDigitTable = charArrayOf('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f')
    return buildString(hash.size * 2) {
        for (b in hash) {
            append(hexDigitTable[b.toInt() shr 4 and 0xF])
            append(hexDigitTable[b.toInt() and 0xF])
        }
    }
}
```

---

Fixed.